plugins/s3_auth/unit_tests/test_aws_auth_v4.cc

```diff
@@ -675,7 +678,42 @@ TEST_CASE("S3AuthV4UtilParams: include all headers by default", "[AWS][auth][uti
     "819a275bbd601fd6f6ba39190ee8299d34fcb0f5e0a4c0d8017c35e79a026579",
   };
 
-  ValidateBenchCanonicalRequest(api, /*signePayload */ false, &now, bench, include, exclude);
+  ValidateBenchCanonicalRequest(api, /*signPayload */ false, &now, bench, include, exclude);
+}
+
+TEST_CASE("S3AuthV4UtilParams: include all headers by default, adding session token", "[AWS][auth][utility]")
+{
+  time_t now = 1369353600; /* 5/24/2013 00:00:00 GMT */
+
+  /* Define the HTTP request elements */
+  MockTsInterface api;
+  api._method.assign("GET");
+  api._host.assign("examplebucket.s3.amazonaws.com");
+  api._path.assign("");
+  api._query.assign("max-keys=2&prefix=J");
+  api._headers["Host"]                 = "examplebucket.s3.amazonaws.com";
+  api._headers["Content-Type"]         = "gzip";
+  api._headers["x-amz-content-sha256"] = "UNSIGNED-PAYLOAD";
+  api._headers["x-amz-date"]           = "20130524T000000Z";
+  api._headers["x-amz-security-token"] = awsSessionToken;
+  api._headers["HeaderA"]              = "HeaderAValue";
+  api._headers["HeaderB"]              = "HeaderBValue";
+  api._headers["HeaderC"]              = "HeaderCValue";
+  api._headers["HeaderD"]              = "HeaderDValue";
+  api._headers["HeaderE"]              = "HeaderEValue";
+  api._headers["HeaderF"]              = "HeaderFValue";
+
+  StringSet include = defaultIncludeHeaders;
+  StringSet exclude = defaultExcludeHeaders;
+
+  const char *bench[] = {
+    /* Signed Headers */
+    "content-type;headera;headerb;headerc;headerd;headere;headerf;host;x-amz-content-sha256;x-amz-date;x-amz-security-token",
+    /* Canonical Request sha256 */
+    "0ae64838a775b635ff5d928e38d0a364e953bc60e668af19fba38cbba905769e",
+  };
+
+  ValidateBenchCanonicalRequest(api, /*signPayload */ false, &now, bench, include, exclude);
 }
```

Comment:
This duplicates the previous test, but adds the new header field.

Reply:
* Remove the added test, because the signing of arbitrary headers is already covered by existing tests.
* Add a test to confirm that the new header is added when the plugin is configured to add it.

Do I have that right?

Reply:
> Remove the added test, because the signing of arbitrary headers is already covered by existing tests.

right.  

> Add a test to confirm that the new header is added when the plugin is configured to add it.

The Catch unit tests test only the v4 calculation. Your changes are in the plugin which will be hard to test from the unit-test we have. I guess you would need to create AuTest test which you will have to be created from scratch for this plugin (it will be great but it is up to you).